trino-lb/src/routing/explain_costs.rs

```diff
@@ -21,7 +28,23 @@ pub struct ExplainCostsRouter {
 
 impl ExplainCostsRouter {
     #[instrument(name = "ExplainCostsRouter::new")]
-    pub fn new(config: &ExplainCostsRouterConfig) -> Result<Self, Error> {
+    pub fn new(
+        config: &ExplainCostsRouterConfig,
+        valid_target_groups: HashSet<String>,
+    ) -> Result<Self, Error> {
```

Comment:
**suggestion (non-blocker)**: you could make a Result type alias so the error doesn't need to be provided each time.

Reply:
While starting this project I intentionally choose against it as it did feel at bit too much magic for me, and now I'm sticking with it for consistency reasons. But I don't have any strong feelings and would be fine to swap it out everywhere

Reply:
Fair enough, though I don't think there's too much magic once you see that it simply sets the error variant so it doesn't need to be specified each time (since we encourage all errors be some variant in the snafu `Error` struct in the same file):

```rust
type Result<T, E = Error> = std::result::Result<T, E>;
```

leaving us to do:

```rust
   pub fn new() -> Result<Self>
```